xrdp/xrdp_mm.c

```diff
@@ -1432,19 +1432,28 @@ xrdp_mm_egfx_caps_advertise(void *user, int caps_count,
                 break;
         }
     }
-    if (best_pro_index >= 0)
+#if defined(XRDP_H264)
+    struct xrdp_tconfig_gfx_codec_order co = self->wm->gfx_config->codec;
+    bool_t use_h264 = (best_h264_index >= 0 && (best_pro_index < 0 || (co.h264_idx >= 0 && co.h264_idx < co.rfx_idx)));
```

Comment:
We have 2 codecs for GFX now. It might be more complex when we support a third codec.